### PR TITLE
Allow providing an additional comment on step method invocations

### PR DESCRIPTION
**Pull request by Airblader**
This PR is not yet complete. As of now, it implements the basic mechanism and printing a comment in the PlainTextReporter. Missing:

* Displaying it in the HTML report.
* Tests, obviously. :-)

I just want to run it by you first to get feedback on the overall solution and also to discuss how the HTML report should display the comment.

### CODE REVIEW

File: jgiven-core/src/main/java/com/tngtech/jgiven/impl/ScenarioModelBuilder.java [hunk 1]
```diff
@@ -141,6 +141,24 @@ public void introWordAdded( String value ) {
...
     }
 
+    @Override
+    public void stepCommentAdded( List<NamedArgument> arguments ) {
+        if( arguments == null || arguments.size() != 1 ) {
+            throw new JGivenWrongUsageException( "A step comment method must have exactly one parameter." );
+        }
+
+        if( !( arguments.get( 0 ).getValue() instanceof String ) ) {
+            throw new JGivenWrongUsageException( "The step comment method parameter must be a string." );
```
Comment (janschaefer) [inline jgiven-core/src/main/java/com/tngtech/jgiven/impl/ScenarioModelBuilder.java]: One could argue here that we could use any object and just use the toString method

**Comment by janschaefer** [comment]
Looks really good so far. The big question is now how to visualize this in the HTML report. One possibility is to show it like the ExtendedDescription with a tooltip. If there is also an extended description that might be combined in a single tooltip, maybe.

**Comment by janschaefer** [comment]
this PR implements #50

**Comment by Airblader** [comment]
@janschaefer I was just wondering how this plays together with `CurrentStep`. Why is `addAttachment` exposed via `CurrentStep`, but `comment` is a direct stage method? Should we reconsider this design?

Also: is there an easy(!) way to generate the HTML report for the tests? :-) I can't seem to get the classpath right when calling it from command-line; when simply calling the generator through a test, I get a NPE in `unzipApp`. :-(

File: jgiven-core/src/main/java/com/tngtech/jgiven/impl/ScenarioModelBuilder.java (hunk 1 above)
Comment (Airblader) [inline jgiven-core/src/main/java/com/tngtech/jgiven/impl/ScenarioModelBuilder.java]: One could indeed. But is there a usecase for it? :-) Seems like a non-string comment is actually more of an attachment.

**Comment by janschaefer** [comment]
As the comment is something which is a feature for ad-hoc commenting steps I think it makes more sense to have that as a Stage method. 

Regarding the test problem. It should work when you add the folder `jgiven-html5-report/build/resources/main` to the test classpath

**Comment by Airblader** [comment]
> As the comment is something which is a feature for ad-hoc commenting steps I think it makes more sense to have that as a Stage method.

Alright, let's proceed then. I managed to hack together a test that creates the HTML output now, so I'll be working on the tooltip and tests.

**Comment by Airblader** [comment]
Here's what it'll look like:

screenshot-2016-06-26_23-57-40
screenshot-2016-06-26_23-57-37
screenshot-2016-06-26_23-57-34

**Comment by janschaefer** [comment]
Looks good :-)

**Comment by janschaefer** [comment]
If you could add a test (does not have to be on the HTML level) we can merge it

**Comment by Airblader** [comment]
@janschaefer And done!

**Comment by janschaefer** [comment]
Thanks!

**Pull request merged**